pkg/network/protocols/http2/protocol.go

```diff
@@ -255,7 +288,46 @@ func (p *protocol) setupMapCleaner(mgr *manager.Manager) {
 		return started > 0 && (now-started) > ttl
 	})
 
-	p.mapCleaner = mapCleaner
+	p.http2InFlightMapCleaner = mapCleaner
+}
+
+func (p *protocol) setupDynamicTableMapCleaner(mgr *manager.Manager) {
+	dynamicTableMap, _, err := mgr.GetMap(dynamicTable)
+	if err != nil {
+		log.Errorf("error getting %q map: %s", dynamicTable, err)
+		return
+	}
+
+	mapCleaner, err := ddebpf.NewMapCleaner(dynamicTableMap, new(http2DynamicTableIndex), new(http2DynamicTableEntry))
+	if err != nil {
+		log.Errorf("error creating map cleaner: %s", err)
+		return
+	}
+
+	var terminatedConnectionsMap map[netebpf.ConnTuple]struct{}
+	mapCleaner.Clean(p.cfg.HTTP2DynamicTableMapCleanerInterval,
+		func() bool {
+			terminatedConnectionsMap = make(map[netebpf.ConnTuple]struct{})
```

Comment:
nit: this will hold the map memory until the next `Clean` runs. It would be better to release it right away after the clean loop is done, yeah?

Reply:
changed